File: 哈希表/128.最长连续序列.py


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # 去重，将查找操作转换为O(1)
        nums = set(nums)
        best = 0
        # 从任意一个存在的连续序列的起点开始 找最长的
        for i in nums:
            if i - 1 not in nums:
                j = i
                # j = i + 1 #少算一次
                while j in nums:
                    j += 1
                best = max(best, j - i)
        return best

File: 技巧/128.最长连续序列.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  let res = 0;
  nums = new Set(nums);
  for (const i of nums) {
    if (!nums.has(i - 1)) {
      j = i;
      while (nums.has(j)) j++;
      res = Math.max(res, j - i);
    }
  }
  return res;
};
